browser/ui/views/frame/brave_browser_view_layout.cc

```diff
@@ -250,14 +254,19 @@ void BraveBrowserViewLayout::UpdateContentsContainerInsets(
   // Control contents's margin with sidebar & vertical tab state.
   gfx::Insets contents_margins = GetContentsMargins();
 
+  // In rounded corners mode, we need to include a little margin so we have
+  // somewhere to draw the shadow.
+  int contents_margin_for_rounded_corners =
+      BraveBrowser::GetRoundedCornersWebViewMargin(browser_view_->browser());
+
   // Don't need contents container's left or right margin with vertical tab as
   // vertical tab itself has sufficient padding.
   if (tabs::utils::ShouldShowVerticalTabs(browser_view_->browser()) &&
       !IsFullscreenForBrowser()) {
     if (tabs::utils::IsVerticalTabOnRight(browser_view_->browser())) {
-      contents_margins.set_right(0);
+      contents_margins.set_right(contents_margin_for_rounded_corners);
```

Comment:
This is where we remove the margin from the webcontents for vertical tabs. This fixes the issue, but it means we double pad @simonhong